docs/data/data-grid/accessibility/GridAriaV7.js

```diff
@@ -2,30 +2,28 @@ import * as React from 'react';
 import { DataGrid, GridToolbar } from '@mui/x-data-grid';
 import { useDemoData } from '@mui/x-data-grid-generator';
 
-const experimentalFeatures = { ariaV7: true };
-const slots = { toolbar: GridToolbar };
-
 export default function GridAriaV7() {
   const { data } = useDemoData({
     dataSet: 'Commodity',
     rowLength: 100,
     maxColumns: 6,
   });
 
-  const initialState = React.useMemo(() => {
-    return {
+  const initialState = React.useMemo(
+    () => ({
       ...data.initialState,
       pagination: { paginationModel: { pageSize: 10 } },
-    };
-  }, [data.initialState]);
+    }),
+    [data.initialState],
+  );
 
   return (
     <div style={{ height: 400, width: '100%' }}>
       <DataGrid
         {...data}
-        slots={slots}
+        slots={{ toolbar: GridToolbar }}
         initialState={initialState}
-        experimentalFeatures={experimentalFeatures}
+        experimentalFeatures={{ ariaV7: true }}
```

Comment:
We've been discussing these notations here: https://github.com/mui/mui-x/pull/10326#discussion_r1324084778

The consensus seemed to be we should use the correct one rather than the easy one.

Reply:
Is it the same case? As I understand it `experimentalFeatures` reference can change between renders (the value needs to stay the same for the screen reader to be coherent) but `getRowId` reference can't change.

Reply:
Yeah so it's not as bad as a prop that triggers a grid logic computation, but changing a prop will still force a re-render of the root grid component. And as far as I remember, the root props are used all the way down, and we don't optimize the use-case where a single field of the props changes yet (see solution to point 1 of #9001); if one field changes, anyone using the props (via `useGridRootProps`) needs to re-render.

This includes the grid cells which are the most expensive part of full re-renders, due to their number:

https://github.com/mui/mui-x/blob/b435ed1a5048f55fb445bfba354e316cb809d4d1/packages/grid/x-data-grid/src/components/cell/GridCell.tsx#L548

The problem is that if anything upstream triggers a re-render, then it's a forced re-render inside the grid, e.g. in the example below every time they type something in the search input, the grid does a full re-render.

```tsx
function C() {
  const [value, setValue] = useState('')
  const { rows, columns } = useSomeQuery(...)
  return (
    <div>
      <input value={value} onChange={e => setValue(e.target.value)} placeholder='Search me' />
      <DataGrid rows={rows} columns={columns} experimentalFeatures={{ ariaV7: true }} />
    </div>
  )
}
```

I'm not sure what's the right answer. I prefer the compact notation for the readability benefits, but if users don't understand how to use react then our docs are not helping them.

Reply:
> changing a prop will still force a re-render of the root grid component.

@romgrk Oh, I see what you mean. It's tricky, UX vs. DX. I believe most of the demos in the docs don't follow https://mui.com/x/react-data-grid/performance/, and yet it helps performance. 

In the past, I wondered if we couldn't make the memorization of props close to irrelevant by better isolating the work done at each render. Maybe the root data grid could only register the root props in a context, they have one memorized child with no props stopping the rendering chain, and lastly have the data grid children under the child triggered when selective parts of the root props change.

Reply:
> I'd be fine with the compromise of this PR:

@romgrk I like the way you clearly phrased it.

- It's aligned with how we have been doing things so far.
- I think it's aligned with how we are likely to improve the state of things going forward. I don't see how we could ever change providing a different `getRowId` or `rows` prop not to recompute the data. However, we might be able to make cheaper re-renders.